pallets/did/src/mock.rs

```diff
@@ -0,0 +1,274 @@
+// KILT Blockchain – https://botlabs.org
+// Copyright (C) 2019-2021 BOTLabs GmbH
+
+// The KILT Blockchain is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The KILT Blockchain is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+// If you feel like getting in touch with us, you can do so at info@botlabs.org
+
+use crate as did;
+use crate::*;
+
+use frame_support::{parameter_types, weights::constants::RocksDbWeight};
+use kilt_primitives::{AccountId, Signature};
+use sp_core::{ed25519, sr25519, Pair, H256};
+use sp_runtime::{
+	testing::Header,
+	traits::{BlakeTwo256, IdentifyAccount, IdentityLookup, Verify},
+};
+
+pub type UncheckedExtrinsic = frame_system::mocking::MockUncheckedExtrinsic<Test>;
+pub type Block = frame_system::mocking::MockBlock<Test>;
+
+frame_support::construct_runtime!(
+	pub enum Test where
+		Block = Block,
+		NodeBlock = Block,
+		UncheckedExtrinsic = UncheckedExtrinsic,
+	{
+		System: frame_system::{Pallet, Call, Config, Storage, Event<T>},
+		Did: did::{Pallet, Call, Storage, Event<T>},
+	}
+);
+
+parameter_types! {
+	pub const SS58Prefix: u8 = 38;
+	pub const BlockHashCount: u64 = 250;
+}
+
+impl frame_system::Config for Test {
+	type Origin = Origin;
+	type Call = Call;
+	type Index = u64;
+	type BlockNumber = u64;
+	type Hash = H256;
+	type Hashing = BlakeTwo256;
+	type AccountId = <<Signature as Verify>::Signer as IdentifyAccount>::AccountId;
+	type Lookup = IdentityLookup<Self::AccountId>;
+	type Header = Header;
+	type Event = ();
+	type BlockHashCount = BlockHashCount;
+	type DbWeight = RocksDbWeight;
+	type Version = ();
+
+	type PalletInfo = PalletInfo;
+	type AccountData = ();
+	type OnNewAccount = ();
+	type OnKilledAccount = ();
+	type BaseCallFilter = ();
+	type SystemWeightInfo = ();
+	type BlockWeights = ();
+	type BlockLength = ();
+	type SS58Prefix = SS58Prefix;
+}
+
+impl did::Config for Test {
+	type Event = ();
+	type WeightInfo = ();
+	type DIDIdentifier = AccountId;
+}
+
+pub type TestDidIdentifier = <Test as did::Config>::DIDIdentifier;
+
+#[allow(dead_code)]
```

Comment:
Were all of these features required for you? I removed them and didn't have any complaints. We should not use them if they are not 100% required.

I'm, on `1.52.0-nightly (f98721f88 2021-03-10)`.

Reply:
What features are you referring to exactly? The exposed utility functions you mean? Or the `#[allow(dead_code)]`?

Reply:
The latter.

Reply:
Ok yes they were there because clippy was complaining when I wrote them down. But then I used them all in the tests and fogot to remove them (clippy doesn't notify you in reverse apparently 🥲 )